qa/tasks/cephfs/test_mds_metrics.py

```diff
@@ -511,3 +511,60 @@ def test_client_metrics_and_metadata(self):
             if not (client_metadata[i]['valid_metrics']):
                 raise RuntimeError("valid_metrics not found!")
 
+    def test_client_metrics_after_failing_existing_filesystem(self):
+        self.mount_a.umount_wait()
+        self.mount_b.umount_wait()
+
+        self.mds_cluster.mon_manager.raw_cluster_cmd("fs", "flag", "set",
+                    "enable_multiple", "true", "--yes-i-really-mean-it")
+
+        # creating filesystem
+        fs_a = self._setup_fs(fs_name="fs1")
+
+        # Mount a client on fs_a
+        self.mount_a.mount_wait(cephfs_name=fs_a.name)
+        self.mount_a.write_n_mb("pad.bin", 1)
+        self.mount_a.write_n_mb("test.bin", 2)
+        self.mount_a.path_to_ino("test.bin")
+        self.mount_a.create_files()
+
+        # creating another filesystem
+        fs_b = self._setup_fs(fs_name="fs2")
+
+        # Mount a client on fs_b
+        self.mount_b.mount_wait(cephfs_name=fs_b.name)
+        self.mount_b.write_n_mb("test.bin", 1)
+        self.mount_b.path_to_ino("test.bin")
+        self.mount_b.create_files()
+
+        # validate
+        valid, metrics = self._get_metrics(
+            self.verify_mds_metrics(client_count=TestMDSMetrics.CLIENTS_REQUIRED), 30)
+        log.debug(f"metrics={metrics}")
+        self.assertTrue(valid)
+
+        self.mount_b.umount_wait(force=True)
+
+        # fail filesystem 'fs2'
+        fs_b.fail()
+
+        time.sleep(10)
+
+        # creating another filesystem
+        fs_c = self._setup_fs(fs_name="fs3")
+
+        # Mount a client on fs_c
+        self.mount_b.mount_wait(cephfs_name=fs_c.name)
+        self.mount_b.write_n_mb("test.bin", 1)
+        self.mount_b.path_to_ino("test.bin")
+        self.mount_b.create_files()
+
+        client_metadata = metrics['client_metadata']
+        global_metrics = metrics['global_metrics']
+
+        client_c_name = f'client.{self.mount_b.get_global_id()}'
+
+        if client_c_name not in client_metadata\
+                or client_c_name not in global_metrics:
+            raise RuntimeError("Client of new filesystem is missing!")
```

Comment:
Good check. But it could be more specific like checking inside `client_metadata[fs_c.name]` ?

Also update the error message like `Invalid metrics after the filesystem {fs_c.name} fail`.

Reply:
Actually, addition of fs_name in `perf stats` is not yet merged in main.So,we have to write it as per our earlier output structure until new changes comes in mainline.

Reply:
That's okay, you can make the changes based on the new `perf stats` output. For local testing, add the other commit in this branch.

Reply:
If that's the case, I have added `mul_fs` field also.